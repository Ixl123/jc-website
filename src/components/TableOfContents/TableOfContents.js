import React from 'react';
import theme from 'styled-theming';

import styled from 'styled-components';
import PropTypes from 'prop-types';
import { FaListUl } from 'react-icons/fa';
import { AnchorLink } from 'gatsby-plugin-anchor-links';

import Divider from '../Divider';
import Heading from '../Heading';

const TableOfContentsColor = theme(`mode`, {
  light: (props) => props.theme.colors.black,
  dark: (props) => props.theme.colors.white,
});

const TableOfContentsBackgroundColor = theme(`mode`, {
  light: (props) => props.theme.colors.light,
  dark: (props) => props.theme.colors.dark,
});

const StyledTableOfContentsNavigation = styled.nav`
  display: none;
  padding: 1.6rem;
  margin-left: 1rem;
  flex-direction: column;
  align-items: center;
  justify-content: flex-start;
  align-items: flex-start;

  max-width: 300px;
  font-family: 'Raleway', sans-serif;
  color: ${TableOfContentsColor};
  background: ${TableOfContentsBackgroundColor};
  position: sticky;
  top: 70px;

  ${(props) => props.theme.xlarge} {
    display: flex;
  }
  h2 {
    font-size: 1.8rem;
    font-weight: var(--font-weight-medium);
    text-transform: uppercase;
    letter-spacing: 0.2rem;
    text-rendering: optimizelegibility;
    margin-bottom: 16px;
  }

  a {
    display: block;
    opacity: 0.8;
    text-decoration: none;
    transition: opacity 500ms ease 0s;
    margin-top: 1rem;
    text-align: left;
  }

  a:hover,
  a:focus {
    color: var(--color-primary);
  }
`;

const TableOfContents = (props) => {
  const { items } = props;
  return (
    <div>
      {items && items.length > 0 ? (
        <StyledTableOfContentsNavigation>
          <Heading appearance={`H2`}>
            <FaListUl></FaListUl> Table of contents
          </Heading>

          <Divider></Divider>
          {items.map((item) => {
            const { url, title } = item;
            return (
              <AnchorLink key={title} to={`./${url}`}>
                {title}
              </AnchorLink>
            );
          })}
        </StyledTableOfContentsNavigation>
      ) : null}
    </div>
  );
};

PropTypes.displayName = `TableOfContents`;

export default TableOfContents;
