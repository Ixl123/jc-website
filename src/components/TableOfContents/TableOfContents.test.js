import React from 'react';
import { render } from '@testing-library/react';

import TableOfContents from './TableOfContents';

const items = [
  {
    url: `#exciting-stuff`,
    title: `Exciting Stuff`,
  },
  {
    url: `#what-i-have-learned-in-a-nutshell`,
    title: `What I have learned in a nutshell`,
  },
  {
    url: `#some-improvement-ideas`,
    title: `Some improvement ideas`,
  },
];

test(`Divider is rendered properly`, () => {
  const { container } = render(<TableOfContents items={items} />);

  expect(container).toMatchSnapshot();
});

test(`TableOfContents can render multiple items`, () => {
  const { getByRole } = render(<TableOfContents items={items} />);
  getByRole(`link`, { name: /some improvement ideas/i });
  getByRole(`link`, { name: /what I have learned in a nutshell/i });
  getByRole(`link`, { name: /exciting Stuff/i });
});
test(`TableOfContents can render no items`, () => {
  const { queryByRole } = render(<TableOfContents items={[]} />);
  expect(queryByRole(`link`)).not.toBeInTheDocument();
});
