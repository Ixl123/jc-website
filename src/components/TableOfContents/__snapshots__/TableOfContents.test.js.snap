// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Divider is rendered properly 1`] = `
.c1 {
  display: block;
  width: 55px;
  height: 12px;
  margin: 0 auto;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: column;
  -ms-flex-direction: column;
  flex-direction: column;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  -webkit-box-pack: start;
  -webkit-justify-content: flex-start;
  -ms-flex-pack: start;
  justify-content: flex-start;
  -webkit-align-items: flex-start;
  -webkit-box-align: flex-start;
  -ms-flex-align: flex-start;
  align-items: flex-start;
  max-width: 400px;
  font-family: 'Raleway',sans-serif;
  color: var(--color-text);
}

.c0 h2 {
  font-size: 1.6rem;
  font-weight: var(--font-weight-medium);
  text-transform: uppercase;
  -webkit-letter-spacing: 0.2rem;
  -moz-letter-spacing: 0.2rem;
  -ms-letter-spacing: 0.2rem;
  letter-spacing: 0.2rem;
  text-rendering: optimizelegibility;
  color: hsl(210deg,25%,88%);
  margin-bottom: 16px;
}

.c0 a {
  display: block;
  opacity: 0.7;
  color: var(--color-gray-800);
  -webkit-text-decoration: none;
  text-decoration: none;
  -webkit-transition: opacity 500ms ease 0s;
  transition: opacity 500ms ease 0s;
  margin-top: 1rem;
  color: black;
  text-align: left;
}

.c0 a:hover,
.c0 a:focus {
  color: var(--color-text);
}

<div>
  <nav
    class="c0"
  >
    <h2>
      Table of contents
    </h2>
    <svg
      class="c1"
    >
      <g
        fill="none"
        fill-rule="evenodd"
        stroke="none"
        stroke-width="1"
      >
        <g
          class=""
          transform="translate(-483.000000, -1059.000000)"
        >
          <g
            transform="translate(182.000000, 936.000000)"
          >
            <g
              transform="translate(301.000000, 123.000000)"
            >
              <rect
                fill-opacity="0.3"
                height="12"
                width="35"
                x="0"
                y="0"
              />
              <rect
                fill-opacity="0.1"
                height="12"
                width="35"
                x="20"
                y="0"
              />
            </g>
          </g>
        </g>
      </g>
    </svg>
    <a
      href="#exciting-stuff"
    >
      Exciting Stuff
    </a>
    <a
      href="#what-i-have-learned-in-a-nutshell"
    >
      What I have learned in a nutshell
    </a>
    <a
      href="#some-improvement-ideas"
    >
      Some improvement ideas
    </a>
  </nav>
</div>
`;
